From 32f74906c124e4bf2c3cf78fa48bad30c9418faf Mon Sep 17 00:00:00 2001
From: MeiChia Chiu <MeiChia.Chiu@mediatek.com>
Date: Wed, 15 May 2024 15:23:55 +0800
Subject: [PATCH 083/210] mtk: hostapd: Fix Operating Mode Notification issue
 in 2GHz

If this patch is not applied, since the driver enables the
Operating Mode Notification feature, hostapd will enable the OMN bit
in the Extended Capabilities Element when constructing beacon.
However, this is not allowed on frequency bands that do not support
ieee80211ac, hence add this patch to fix this issue.

Signed-off-by: MeiChia Chiu <MeiChia.Chiu@mediatek.com>
---
 src/ap/ieee802_11_shared.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/ap/ieee802_11_shared.c b/src/ap/ieee802_11_shared.c
index 28c2ccb1c..18f65c4e4 100644
--- a/src/ap/ieee802_11_shared.c
+++ b/src/ap/ieee802_11_shared.c
@@ -521,6 +521,11 @@ u8 * hostapd_eid_ext_capab(struct hostapd_data *hapd, u8 *eid,
 			*pos &= ~0x08;
 		if (i == 2 && !hapd->iconf->mbssid)
 			*pos &= ~0x40;
+
+		/* Clear bits 62 (Operating Mode Notification)
+		 * if ieee80211ac is not enabled (mainly 2.4G and 6G) */
+		if (i == 7 && !hapd->iconf->ieee80211ac)
+			*pos &= ~0x40;
 	}
 
 	while (len > 0 && eid[1 + len] == 0) {
-- 
2.45.2

